File: DSA/stack/javascript/reverseString.js
#!/bin/node

const Stack = require("./stack");

/**
 * Implementation of Reverse String Algorithm.
 */

function reverseString(inputString) {
  let stack = new Stack();
  let reverseStr = "";

  for (let i = 0; i < inputString.length; i++) {
    stack.push(inputString[i]);
  }

  while (!stack.isEmpty()) {
    reverseStr += stack.pop();
  }

  return reverseStr;
}

let str1 = "!evitacudE ot emocleW";
let str2 = "Hello, World! from Reverse String Function.";
console.log(reverseString(str1));
console.log(reverseString(str2));

File: DSA/Stack/python/reverse_string.py
#!/bin/python3

"""
    Implementation of Reverse String Algorithm.
"""

from stack import Stack

def reverse_string(input_str):
    """Implementation of the Function."""
    stack = Stack()
    reversed_str = ""

    for i in range(len(input_str)):
        stack.push(input_str[i])

    while not stack.is_empty():
        reversed_str += stack.pop()

    return reversed_str

input_str1 = "!evitacudE ot emocleW"
print(reverse_string(input_str1))
input_str2 = "Hello, World! From Reverse"
print(reverse_string(input_str2))
